### DISCUSSION TITLE
What is the best workflow to sync two TODO.TXTs edited on two different devices?

### DISCUSSION THREAD

**Discussion by tempuserr** [ransome1/sleek#727] [Q&A]
On two different devices (PC and Android) I use 2 different apps to maintain Task List (Sleek on PC and Simpletask on Android).

What is the best way to sync (merge) this todo.txt Lists when edits (e.g. adding new tasks, completion others, changing due dates, deleting, etc.) ale done on both of them independently?
Is it possible for this kind of two-way sync? 
What is the best automatic method or software to do this?

**Comment by jbtodo** [comment]
Is this an ongoing sync between different files or do you just want to sync them up once and then use a single todo.txt file?

For the second case, I have had good luck with Dropbox (free account) using Sleek and SwiftoDo on iOS (not sure what is the comparable Android app). I can edit on both (just not withing seconds of each other... and they sync up just fine.
If it is a onetime sync, that is the beauty of txt files, you can just copy paste everything together and then clean it up on your preferred platform.

**Reply by tempuserr** [reply]
I mean ongoing sync.
I adding task on my Android phone when on a road and on Windows when working at computer.
So my task gets different on both lists - some can be common, but another tasks are only on one of this two lists.

There is an article which shows what is a main issue with todo.txt: https://rtalbert.org/back-to-todoist/
Unfortunately I agree with that - there is not a reliable ongoing two-way sync of todo.txts at the moment. As far I have researched for it.

Of course it is possible to make it - comparing two text files is not a rocket science, but this needs to be built into clients or added to some kind of network/server service which merge two lists when clients connects to it. So this is task rather to senior programmer.

**Reply by msdisme** [reply]
I appreciate the work being done on Sleek - the pm: extension is interesting and I appreciate a lot of the UI.   I use another todo.txt tool instead (well, also, it's an extremely stable great tool) because it solved syncing todo.txt files back in 2018 or 2019.

**Reply by msdisme** [reply]
@ransome1 just realized after edits I cut the details on how they sync - https://swiftodoapp.com/troubleshooting/file-sync/.  I won't say I never see a problem, but it's once in a blue moon.

**Comment by mshuflin** [comment]
I use synching for this

**Reply by tempuserr** [reply]
Syncthing can merge different lines from two different text files and merge it into one??

**Reply by mshuflin** [reply]
If you have syncthing syncing the whole time, it does not need to merge anything

**Reply by tempuserr** [reply]
Maybe it will work for people with all devices permanently turned on and online, who don't leave one big city.
As described in above linked article it will not work in other cases (traveling, flights, mountain trips, etc.) for others, and I'm in this group.

But thanks for a tip, maybe this would be a solution for users from first group.

**Reply by ERamseth** [reply]
I also use syncthing. 

In your use case I think you’ll end up with duplicate files if edited “at the same time” on two devices. How often is this happening though?

if you have syncthing let’s say on desktop at home, desktop at office, and mobile device with you on a plane. Edit files on mobile devices with not internet. Get off plane, connect to internet, your edits will go back to the syncthing and you replace the older files. 

If for some reason you proceed to edit on home computer before mobile devices can sync (battery is dead, let’s say). Then later turn the mobile device back on and connect to internet. Then when it does sync you will get conflict files as described here:

https://docs.syncthing.net/users/syncing.html (see the section called “conflicting changes”).

Once you have the conflict files you would need to merge them manually (or with some tool). 

I guess you can probably also use something like git to serve as a sync repository and to merge files.
